flink-ai-flow/wiki_doc/Tutorial.md

```diff
@@ -0,0 +1,383 @@
+
+
+## Configuration of Flink AI Flow Server
+To work with Flink AI Flow, we should add some configurations for Flink AI Flow Server.
+The config file, named as `master.yaml`, should locate in `$AIRFLOW_HOME` or `~/airflow` directory.
+
+Remember to update fields according to your own environment.
+
+### master.yaml
+```yaml
+ # Config of master server
+# endpoint of AI Flow Server
+server_ip: localhost
+server_port: 50051
+# uri of database backend of AIFlow server
+db_uri: {db_uri} # sqlite:////path/to/airflow/aiflow.db or mysql://username:password@127.0.0.1/airflow
+# type of database backend in master
+db_type: {db_type} # sql_lite or mysql
+# whether to start the scheduler service
+start_scheduler_service: True
+# uri of notification service
+notification_uri: localhost:50052
+# scheduler config
+scheduler:
+  scheduler_class_name: ai_flow_plugins.scheduler_plugins.airflow.airflow_scheduler.AirFlowScheduler
+  scheduler_config:
+    airflow_deploy_path: /absolute/path/to/airflow/airflow_deploy
+    notification_service_uri: localhost:50052
+```
+
+## Prepare Project Directory
+Before writing the workflow, we should prepare a project directory whose structure is as follows:
+
+```
+tutorial_project/
+        |- workflows/
+           |- tutorial_workflow/
+              |- tutorial_workflow.py 
+              |- tutorial_workflow.yaml 
+        |- dependencies/
+            |-python 
+            |-jar
+        |- resources/
+        └─ project.yaml
+```
+
+`tutorial_project` is the root directory of the project and `workflows` is used to save codes and config files of workflows in this project. In this tutorial, we will create a `tutorial_workflow` directory for our workflow. In the `tutorial_workflow.py`, we will add our actual codes for defining the workflow and `tutorial_workflow.yaml` includes some configs of our workflow.
+
+The `dependecies` directory is used to save python/java/go dependencies that will be used by our workflow.
+
+The `resources` directory is for saving all other files(e.g., datasets) that will be used by the project.
+
+The `project.yaml` is for configuring the whole project. 
+
+Now let's go over above config files one by one.
+
+### project.yaml
+Here is an example of the project.yaml for tutorial project.
+```yaml
+project_name: tutorial_project
+server_ip: localhost
+server_port: 50051
+blob:
+  blob_manager_class: ai_flow_plugins.blob_manager_plugins.local_blob_manager.LocalBlobManager
+```
+In `project_name`, we define the project name, which will be the default namespace of workflows in this project as well. 
+
+> Namespace in Flink AI Flow is used for isolation. Each workflow can only send events to its own namespace while it can listen on multiple namespaces. The reason for enabling listening on multiple namespaces is that the workflow could be triggered by external events from notification service.
+
+For `server_ip` and `server_port`,  they tell where the `AIFlowServer` will starts on.
```

Comment:
they tell where the `AIFlowServer` is running on.

Reply:
fixed